sled-agent/src/services.rs

```diff
@@ -1618,18 +1618,82 @@ impl ServiceManager {
                 zone:
                     OmicronZoneConfig {
                         zone_type: OmicronZoneType::ClickhouseServer { .. },
-                        underlay_address: _,
+                        underlay_address,
                         ..
                     },
                 ..
             }) => {
-                // We aren't yet deploying this service
-                error!(
-                    &self.inner.log,
-                    "Deploying ClickhouseServer zones is not yet supported"
-                );
+                let Some(info) = self.inner.sled_info.get() else {
+                    return Err(Error::SledAgentNotReady);
+                };
+
+                let listen_addr = *underlay_address;
+                let listen_port = &CLICKHOUSE_PORT.to_string();
```

Comment:
I feel like this `&` is misplaced. Do we not get a clippy lint about an immediate-deref or something similar?

Reply:
No clippy warnings 🤷‍♀️ , but will remove